### DISCUSSION TITLE
Running .exe file

### DISCUSSION THREAD

**Discussion by bflo96** [OpenModelica/OpenModelica#12343] [Q&A] [answered]
Hello everyone,

I'm trying to run the .exe file of my model, which I am aware it is available on C:\Users\.....\AppData\Local\Temp\OpenModelica\OMEdit\

However, when I try to run it, it gives me this error:

> The code execution cannot proceed because libSimulationRuntimeC.dll was not found. Reinstalling the program may fix this problem.

After a quick search, I figured **libSimulationRuntimeC.dll** can be found on C:\Program Files\OpenModelica1.22.3-64bit\bin.

Should it be located somewhere else to be able to run?
Would it be possible for an user who doesn't have OpenModelica installed to run the .exe file?

Thank you.

**Comment by AnHeuermann** [comment, marked as answer]
Run the .bat file instead of the .exe file. It will update the `PATH` variable, so that all the OpenModelica DLLs can be loaded.

If you want something that you can share with others check out the Functional Mock-up Interface: https://fmi-standard.org/
OpenModelica can [export](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/fmitlm.html#fmi-export) self-contained FMUs that every FMU importing tool can simulate.

**Reply by casella** [reply]
I guess we should write this in the User's Guide. I'll see what I can do 😅

**Reply by AnHeuermann** [reply]
I'm not sure. People ask about this from time to time. But usually they do when they want to share a simulation executable with someone else or run it in some other program. And that's not the right way to do that.
Our simulation interface isn't stable, but the FMU interface very much is. And we did spend some effort on generating FMUs that are portable.
So I would suggest to point them to ME FMU export instead of explaining how the .bat file works.

Edit: Probably this was your plan all along and I misunderstood^^

**Reply by bflo96** [reply]
I'll check the Functional Mock-Up Interface. My idea is to have sort of a "black-box" system with inputs that can be read from a .TXT file or a .JSON file and then having the relevent outputs coming out in the same format. I was thinking of using C#, but another friend suggested Python instead, so I might actually look more into this subject as well.

I tested the .bat file and it worked just fine for me. It didn't work for a friend who didn't have OpenModelica installed.

Thank you so much for your help.
This is quite a new subject for me and I'm still learning how to navigate through it. Your help is incredible.

**Reply by casella** [reply]
@AnHeuermann each choices has pro's and con's. FMI is great, but if you want to use it stand-alone, it has to be FMI-CS, which honestly is still not on a par with regular OMC simulations at the current development status. I would just state the facts and let people decide.

@bflo96 if you are using Python and you have no problems at having OMC installed, you can drive the whole thing from Python using the OMPython interface. See e.g. this nice paper to see a non-trivial applicaton of that concept: http://dx.doi.org/10.3384/ecp204293
